pkg/v1/tkg/tkgpackageclient/image_utils.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tkgpackageclient
+
+import (
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	dockerParser "github.com/novln/docker-parser"
+	"github.com/pkg/errors"
+
+	kappipkg "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgpackagedatamodel"
+)
+
+// packageRepositoryStdout is used for unmarshal the package repository stdout to get the tag in use
+type packageRepositoryStdout struct {
+	APIVersion  string `yaml:"apiVersion,omitempty"`
+	Directories []struct {
+		Contents []struct {
+			ImgpkgBundle struct {
+				Tag string `yaml:"tag,omitempty"`
+			} `yaml:"imgpkgBundle,omitempty"`
+		} `yaml:"contents,omitempty"`
+	} `yaml:"directories,omitempty"`
+}
+
+// parseRegistryImageURL parses the registry image URL to get repository and tag, tag is empty if not specified
+func parseRegistryImageURL(imgURL string) (repository, tag string, err error) {
+	ref, err := dockerParser.Parse(imgURL)
+	if err != nil {
+		return "", "", err
+	}
+
+	tag = ref.Tag()
+	// dockerParser sets the tag to "latest" if not specified, however we want it to be empty
+	if tag == tkgpackagedatamodel.DefaultRepositoryImageTag && !strings.HasSuffix(imgURL, ":"+tkgpackagedatamodel.DefaultRepositoryImageTag) {
+		tag = ""
+	}
+	return ref.Repository(), tag, nil
+}
+
+// GetCurrentRepositoryAndTagInUse fetches the current tag used by package repository, taking tagselection into account
+func GetCurrentRepositoryAndTagInUse(pkgr *kappipkg.PackageRepository) (repository, tag string, err error) {
+	if pkgr.Spec.Fetch == nil || pkgr.Spec.Fetch.ImgpkgBundle == nil {
+		return "", "", errors.New("failed to find OCI registry URL")
+	}
+
+	repository, tag, err = parseRegistryImageURL(pkgr.Spec.Fetch.ImgpkgBundle.Image)
+	if err != nil {
+		return "", "", errors.Wrap(err, "failed to parse OCI registry URL")
+	}
+
+	if pkgr.Spec.Fetch.ImgpkgBundle.TagSelection != nil && pkgr.Spec.Fetch.ImgpkgBundle.TagSelection.Semver != nil && pkgr.Status.Fetch != nil {
```

Comment:
keep in mind that stdout contents might shift over time (this is why its contents are not exposed as first class fields). if you are planning to parse it, you have to have a fallback. throwing examples of cases:
- apiVersion in output might be different
- directories field might change to be different (might have multiple directories, etc.)

in general i would recommend not to expose this information back to the user. if they need it they could evaluate it themselves.

Reply:
cc @vijaykatam @maralavi 
Do we want to show which tag is being used currently?
Or is it good enough to put something general like `latest release tag`?

Reply:
+1 to what @cppforlife said, stdout is internal detail of package repository object. Unless explicitly exposed through an API object in status lets show generic detail.